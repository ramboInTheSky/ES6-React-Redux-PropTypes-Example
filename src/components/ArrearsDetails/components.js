import styled from 'styled-components';
import { Button, Typography, mediaQueries } from 'nhh-styles';

export const Heading = styled(Typography.H3)`
  margin-bottom: 1.5rem;
`;

export const Label = styled.span`
  font-weight: 700;
  ${mediaQueries.desktop`
    white-space: nowrap;
  `};
  ${mediaQueries.tablet`
    white-space: nowrap;
  `};
`;

export const LabelLink = styled(Button)`
  font-weight: 700;
  font-size: 18px;
  padding: 0;
  text-align: left;
`;

export const Wrapper = styled.div`
  margin-bottom: 1.5rem;
`;
