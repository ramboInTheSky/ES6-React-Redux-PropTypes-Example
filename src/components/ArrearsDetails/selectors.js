import React, { Fragment } from 'react';
import format from 'string-format';
import { formatting } from 'nhh-styles';

import { installmentArrangement } from '../../util/paymentPlan';
import { Label } from './components';

export const getPauseDetails = (pause, labels, type) => {
  if (pause) {
    return (
      <Fragment>
        {format(labels.pauses[type === 'draft' ? 'draftPausePeriod' : 'pausePeriod'], {
          from: formatting.formatDate(pause.startDate),
          to: formatting.formatDate(pause.expiryDate),
        })}
        {type !== 'draft' && pause.isExtended ? (
          <Label>{` ${labels.pauses.isExtended}`}</Label>
        ) : null}
      </Fragment>
    );
  }
  return labels.none;
};

export const getPaymentPlanDetails = (plan, labels) => (
  <Fragment>
    <Label>{labels.paymentPlan.startDate}</Label> {formatting.formatDate(plan.startDate)}
    <br />
    <Label>{labels.paymentPlan.endDate}</Label> {formatting.formatDate(plan.endDate)}
    <br />
    <Label>{labels.paymentPlan.installment}</Label>{' '}
    {installmentArrangement(
      labels.installmentArrangementFormat,
      plan.installment.amount,
      plan.installment.period,
      formatting.formatDate(plan.startDate),
      plan.installment.schedule
    )}
    <br />
    <Label>{labels.paymentPlan.type}</Label> {plan.type}
  </Fragment>
);
