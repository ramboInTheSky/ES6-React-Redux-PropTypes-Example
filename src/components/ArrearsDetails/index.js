import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import format from 'string-format';
import { formatting, Table, tap } from 'nhh-styles';
import has from 'ramda/src/has';

import { getPauseDetails, getPaymentPlanDetails } from './selectors';
import { Label, LabelLink, Wrapper } from './components';
import { Heading } from '../';
import getInternalRoutes from '../../constants/internalRoutes';
import { DRAFT, ACTIVE } from '../../constants/pause';

const getTableData = (labels, props) => {
  const {
    arrearsLength,
    createdOn,
    owner,
    pauseSummary,
    paymentPlan,
    referralSummaries,
    rentDetail,
    status,
    workwiseReferenceNumber,
  } = props;
  const internalRouter = getInternalRoutes({ ...props });

  const PaymentPlanLabel = paymentPlan ? LabelLink : Label;
  const pause = pauseSummary.find(item => item.status === ACTIVE);
  const draftPause = pauseSummary.find(item => item.status === DRAFT);
  const PauseLabel = pause ? LabelLink : Label;
  const DraftPauseLabel = draftPause ? LabelLink : Label;
  const isReferralSummaries = referralSummaries && referralSummaries.length;
  const latestReferralSummary =
    isReferralSummaries && referralSummaries[referralSummaries.length - 1];

  const hasRentProp = property =>
    tap(rentDetail && has(property, rentDetail), () => rentDetail[property]);

  const returnArray = [
    [
      <Label>{labels.patchOwner}</Label>,
      <span data-bdd="ArrearsDetails-patchOwner">{owner.name}</span>,
    ],
    [<Label>{labels.status}</Label>, <span data-bdd="ArrearsDetails-status">{status}</span>],
    [
      <Label>{labels.workwiseReference}</Label>,
      <span data-bdd="ArrearsDetails-workwiseReference">{workwiseReferenceNumber}</span>,
    ],
    [
      <Label>{labels.lengthOfArrears}</Label>,
      <span data-bdd="ArrearsDetails-lengthOfArrears">
        {format(labels.lengthOfArrearsValue, {
          days: arrearsLength,
          date: formatting.formatDate(createdOn),
        })}
      </span>,
    ],
    [
      <Label>{labels.rentDetails.paymentReferenceNumber}</Label>,
      <span data-bdd="ArrearsDetails-paymentReferenceNumber">
        {hasRentProp('paymentReference') || labels.none}
      </span>,
    ],
    [
      <Label>
        {format(labels.rentDetails.collectionRate.label, {
          value: labels.rentDetails.collectionRate.value5w,
        })}
      </Label>,
      <span data-bdd="ArrearsDetails-paymentCollectionRate5w">
        {tap(hasRentProp('collectionRate5Weeks'), val =>
          format(labels.rentDetails.collectionRate.value, {
            value: val,
          })
        ) || labels.none}
      </span>,
    ],
    [
      <Label>
        {format(labels.rentDetails.collectionRate.label, {
          value: labels.rentDetails.collectionRate.value4m,
        })}
      </Label>,
      <span data-bdd="ArrearsDetails-paymentCollectionRate4m">
        {tap(hasRentProp('collectionRate4Months'), val =>
          format(labels.rentDetails.collectionRate.value, {
            value: val,
          })
        ) || labels.none}
      </span>,
    ],
    [
      <Label>{labels.rentDetails.grossRent.label}</Label>,
      <span data-bdd="ArrearsDetails-grossPayment">
        {tap(hasRentProp('grossRent'), formatting.formatCurrency) || labels.none}
      </span>,
    ],
    ...(tap(hasRentProp('charges'), charges =>
      charges.map(charge => [
        <Label>{charge.description}</Label>,
        <span data-bdd={`ArrearsDetails-charge-${charge.description}`}>
          {formatting.formatCurrency(charge.chargeValue)}
        </span>,
      ])
    ) || []),
    [
      <Label>{labels.rentDetails.paymentFrequency}</Label>,
      <span data-bdd="ArrearsDetails-paymentReferenceNumber">
        {rentDetail ? rentDetail.paymentFrequency : labels.none}
      </span>,
    ],
    [
      <Label>{labels.rentDetails.arrearsBalance}</Label>,
      <span data-bdd="ArrearsDetails-arrearsCurrentBalance">
        {tap(hasRentProp('currentBalance'), val =>
          formatting.formatCurrency(val === 0 ? 0 : val * -1)
        ) || labels.none}
      </span>,
    ],
    [
      <PaymentPlanLabel isText to={internalRouter.paymentPlan}>
        {labels.paymentPlan.label}
      </PaymentPlanLabel>,
      <span data-bdd="ArrearsDetails-paymentPlan">
        {paymentPlan ? getPaymentPlanDetails(paymentPlan, labels) : labels.none}
      </span>,
    ],
    [
      <PauseLabel isText to={internalRouter.pause}>
        {labels.pauses.label}
      </PauseLabel>,
      <span data-bdd="ArrearsDetails-pauses">{getPauseDetails(pause, labels)}</span>,
    ],
    [
      <DraftPauseLabel isText to={internalRouter.pause}>
        {labels.pauses.draftLabel}
      </DraftPauseLabel>,
      <span data-bdd="ArrearsDetails-draftPauses">
        {getPauseDetails(draftPause, labels, 'draft')}
      </span>,
    ],
    [
      <Label>{labels.referralStatus.label}</Label>,
      <span data-bdd="ArrearsDetails-referralStatus">
        {latestReferralSummary ? (
          <Fragment key={`${latestReferralSummary.teamName}_${latestReferralSummary.createdOn}`}>
            {format(labels.referralStatus.value, {
              team: latestReferralSummary.teamName,
              date: formatting.formatDate(latestReferralSummary.createdOn),
            })}
            <br />
          </Fragment>
        ) : (
          labels.none
        )}
      </span>,
    ],
  ];

  return returnArray;
};

const ArrearsDetails = ({ heading, labels, ...props }) => (
  <Wrapper>
    <Heading>{heading}</Heading>
    <Table
      addExtraTd={false}
      addExtraTh={false}
      data-bdd="ArrearsDetailsTable"
      data={getTableData(labels, props)}
      striped
      tdWrap
      topAlign
    />
  </Wrapper>
);

export const detailsPropTypes = {
  createdOn: PropTypes.string.isRequired,
  owner: PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string,
    type: PropTypes.string,
  }),
  pauseSummary: PropTypes.arrayOf(
    PropTypes.shape({
      expiryDate: PropTypes.string.isRequired,
      startDate: PropTypes.string.isRequired,
      status: PropTypes.string.isRequired,
    })
  ),
  paymentPlan: PropTypes.shape({
    endDate: PropTypes.string.isRequired,
    installment: PropTypes.shape({
      amount: PropTypes.number.isRequired,
      period: PropTypes.number.isRequired,
      schedule: PropTypes.string.isRequired,
    }).isRequired,
    startDate: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  }),
  referralSummaries: PropTypes.arrayOf(
    PropTypes.shape({
      createdOn: PropTypes.string.isRequired,
      teamName: PropTypes.string.isRequired,
    })
  ),
  status: PropTypes.string.isRequired,
  arrearsLength: PropTypes.number,
  workwiseReferenceNumber: PropTypes.string,
};

export const labelPropTypes = {
  lengthOfArrears: PropTypes.string.isRequired,
  none: PropTypes.string.isRequired,
  patchOwner: PropTypes.string.isRequired,
  pauses: PropTypes.shape({
    active: PropTypes.string.isRequired,
    draftLabel: PropTypes.string.isRequired,
    draftPausePeriod: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    pausePeriod: PropTypes.string.isRequired,
    pendingApproval: PropTypes.string.isRequired,
  }).isRequired,
  paymentPlan: PropTypes.shape({
    endDate: PropTypes.string.isRequired,
    frequency: PropTypes.string.isRequired,
    instalmentAmount: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    period: PropTypes.string.isRequired,
    startDate: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  }).isRequired,
  referralStatus: PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  }).isRequired,
  status: PropTypes.string.isRequired,
  workwiseReference: PropTypes.string.isRequired,
};

ArrearsDetails.defaultProps = {
  paymentPlan: null,
  pauseSummary: [],
};

ArrearsDetails.propTypes = {
  ...detailsPropTypes,
  heading: PropTypes.string.isRequired,
  labels: PropTypes.shape(labelPropTypes).isRequired,
};

export default ArrearsDetails;
