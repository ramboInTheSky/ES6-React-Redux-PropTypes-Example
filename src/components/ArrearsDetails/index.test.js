import React from 'react';
import { shallow, mount } from 'enzyme';

import ArrearsDetails from './';

describe('<ArrearsDetails />', () => {
  let props;

  beforeEach(() => {
    props = {
      arrearsLength: 5,
      rentDetail: {},
      createdOn: '2018-07-15T13:12:07.193Z',
      heading: 'Arrears details',
      labels: {
        installmentArrangementFormat: '{amount} Every {period} {schedule}(s)',
        rentDetails: {
          paymentReferenceNumber: 'Payment reference number',
          paymentFrequency: 'Debit frequency',
          collectionRate: {
            label: 'Collection rate {value}',
            value5w: '(5W)',
            value4m: '(4M)',
            value: '{value}%',
          },
          grossRent: {
            label: 'Gross rent',
          },
          serviceCharge: {
            label: 'Service charge',
          },
          parkingCharge: 'Parking charge',
          debitFrequency: 'Debit frequency',
          directDebit: 'direct debit',
          recurringCardPayment: 'recurring card payment',
          lastPaymentMethod: 'Last Payment method',
          monthly: 'MONTHLY',
        },
        lengthOfArrears: 'Length of arrears',
        lengthOfArrearsValue: '{days} days (case started on {date})',
        none: 'None',
        patchOwner: 'Patch owner',
        pauses: {
          active: '(active)',
          draftLabel: 'Draft pauses',
          draftPausePeriod: 'Requested from {from} to {to}',
          label: 'Pauses',
          pausePeriod: 'Paused from {from} to {to}',
          pendingApproval: '(pending approval)',
          isExtended: '(extended)',
        },
        paymentPlan: {
          endDate: 'End date:',
          frequency: 'Frequency:',
          instalmentAmount: 'Instalment amount:',
          label: 'Payment plan',
          period: 'Period:',
          startDate: 'Start date:',
          type: 'Type:',
        },
        referralStatus: {
          label: 'Referrals',
          value: '{team}, {date}',
        },
        status: 'Arrears status',
        workwiseReference: 'Workwise reference',
      },
      owner: {
        name: 'Fred',
      },
      phases: [{ name: 'Phase 1' }, { name: 'Phase 2', current: true }],
      status: 'Active',
      workwiseReferenceNumber: '12345',
      legalReferral: {
        title: 'legalReferral',
        createdOn: '2018-07-15T13:12:07.193Z',
      },
      primaryLegalAction: {
        title: 'primaryLegalAction',
        createdOn: '2017-04-23T13:12:07.193Z',
      },
      refferalPack: {
        legalReferralPack: {
          link: '/alink',
        },
      },
    };
  });

  it('renders correctly with minimal props', () => {
    expect(shallow(<ArrearsDetails {...props} />)).toMatchSnapshot();
  });

  it('renders correctly with pauseSummary', () => {
    props.pauseSummary = [
      {
        expiryDate: '2017-04-23T13:12:07.193Z',
        startDate: '2018-07-15T13:12:07.193Z',
        status: 'Active',
      },
    ];
    expect(shallow(<ArrearsDetails {...props} />)).toMatchSnapshot();
  });

  it('renders correctly with paymentPlan', () => {
    props.paymentPlan = {
      endDate: '2017-04-23T13:12:07.193Z',
      installment: {
        amount: 50,
        period: 2,
        schedule: 'Weekly',
      },
      startDate: '2018-07-15T13:12:07.193Z',
      type: 'Foo',
    };
    expect(shallow(<ArrearsDetails {...props} />)).toMatchSnapshot();
  });

  it('renders correctly with referralSummaries', () => {
    props.referralSummaries = [
      {
        createdOn: '2018-07-15T13:12:07.193Z',
        teamName: 'Jim',
      },
      {
        createdOn: '2017-04-23T13:12:07.193Z',
        teamName: 'Bob',
      },
    ];
    expect(shallow(<ArrearsDetails {...props} />)).toMatchSnapshot();
  });

  describe('rent details', () => {
    it('renders correctly with rent details', () => {
      const p = {
        ...props,
        rentDetail: {
          paymentReference: '12039',
          collectionRate5Weeks: 11,
          collectionRate4Months: 10,
          grossRent: 123,
          charges: [
            {
              chargeCode: 'SERV',
              chargeValue: 5,
              chargeDescription: 'Hello world',
            },
          ],
          paymentFrequency: 'Weekly',
          paymentMethod: 'Direct Debit',
          currentBalance: 1230,
        },
      };
      expect(shallow(<ArrearsDetails {...p} />)).toMatchSnapshot();
    });

    it('renders 0 just fine as a collection rate %', () => {
      const p = {
        ...props,
        rentDetail: {
          collectionRate5Weeks: 0,
        },
      };

      const mounted = mount(<ArrearsDetails {...p} />);
      expect(mounted.find({ 'data-bdd': 'ArrearsDetails-paymentCollectionRate5w' }).text()).toEqual(
        '0%'
      );
    });

    it('renders -1 if current balance is 1', () => {
      const p = {
        ...props,
        rentDetail: {
          currentBalance: -1,
        },
      };

      const mounted = mount(<ArrearsDetails {...p} />);
      expect(mounted.find({ 'data-bdd': 'ArrearsDetails-arrearsCurrentBalance' }).text()).toEqual(
        '£1.00'
      );
    });

    it('renders 0 for current balance instead of -0', () => {
      const p = {
        ...props,
        rentDetail: {
          currentBalance: 0,
        },
      };

      const mounted = mount(<ArrearsDetails {...p} />);
      expect(mounted.find({ 'data-bdd': 'ArrearsDetails-arrearsCurrentBalance' }).text()).toEqual(
        '£0.00'
      );
    });
  });
});
