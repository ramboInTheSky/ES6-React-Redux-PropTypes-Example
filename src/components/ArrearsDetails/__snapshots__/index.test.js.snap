// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrearsDetails /> renders correctly with minimal props 1`] = `
<styled.div>
  <Styled(styled.h3)>
    Arrears details
  </Styled(styled.h3)>
  <Table
    addExtraTd={false}
    addExtraTh={false}
    data={
      Array [
        Array [
          <styled.span>
            Patch owner
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-patchOwner"
      >
            Fred
      </span>,
        ],
        Array [
          <styled.span>
            Arrears status
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-status"
      >
            Active
      </span>,
        ],
        Array [
          <styled.span>
            Workwise reference
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-workwiseReference"
      >
            12345
      </span>,
        ],
        Array [
          <styled.span>
            Length of arrears
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-lengthOfArrears"
      >
            5 days (case started on 15/07/2018)
      </span>,
        ],
        Array [
          <styled.span>
            Payment reference number
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentReferenceNumber"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Collection rate (5W)
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentCollectionRate5w"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Collection rate (4M)
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentCollectionRate4m"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Gross rent
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-grossPayment"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Debit frequency
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentReferenceNumber"
      />,
        ],
        Array [
          <styled.span />,
          <span
            data-bdd="ArrearsDetails-arrearsCurrentBalance"
      >
            None
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/payment-plan/create"
      >
            Payment plan
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentPlan"
      >
            None
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/pause-arrears-case/create"
      >
            Pauses
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-pauses"
      >
            None
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/pause-arrears-case/create"
      >
            Draft pauses
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-draftPauses"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Referrals
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-referralStatus"
      >
            None
      </span>,
        ],
      ]
    }
    data-bdd="ArrearsDetailsTable"
    headerData={null}
    striped={true}
    tdWrap={true}
    topAlign={true}
  />
</styled.div>
`;

exports[`<ArrearsDetails /> renders correctly with pauseSummary 1`] = `
<styled.div>
  <Styled(styled.h3)>
    Arrears details
  </Styled(styled.h3)>
  <Table
    addExtraTd={false}
    addExtraTh={false}
    data={
      Array [
        Array [
          <styled.span>
            Patch owner
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-patchOwner"
      >
            Fred
      </span>,
        ],
        Array [
          <styled.span>
            Arrears status
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-status"
      >
            Active
      </span>,
        ],
        Array [
          <styled.span>
            Workwise reference
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-workwiseReference"
      >
            12345
      </span>,
        ],
        Array [
          <styled.span>
            Length of arrears
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-lengthOfArrears"
      >
            5 days (case started on 15/07/2018)
      </span>,
        ],
        Array [
          <styled.span>
            Payment reference number
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentReferenceNumber"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Collection rate (5W)
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentCollectionRate5w"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Collection rate (4M)
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentCollectionRate4m"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Gross rent
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-grossPayment"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Debit frequency
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentReferenceNumber"
      />,
        ],
        Array [
          <styled.span />,
          <span
            data-bdd="ArrearsDetails-arrearsCurrentBalance"
      >
            None
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/payment-plan/create"
      >
            Payment plan
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentPlan"
      >
            None
      </span>,
        ],
        Array [
          <Styled(WithTheme(Button))
            isText={true}
            to="/arrears-details/undefined/pause-arrears-case/create"
      >
            Pauses
      </Styled(WithTheme(Button))>,
          <span
            data-bdd="ArrearsDetails-pauses"
      >
            <Unknown>
                  Paused from 15/07/2018 to 23/04/2017
            </Unknown>
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/pause-arrears-case/create"
      >
            Draft pauses
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-draftPauses"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Referrals
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-referralStatus"
      >
            None
      </span>,
        ],
      ]
    }
    data-bdd="ArrearsDetailsTable"
    headerData={null}
    striped={true}
    tdWrap={true}
    topAlign={true}
  />
</styled.div>
`;

exports[`<ArrearsDetails /> renders correctly with paymentPlan 1`] = `
<styled.div>
  <Styled(styled.h3)>
    Arrears details
  </Styled(styled.h3)>
  <Table
    addExtraTd={false}
    addExtraTh={false}
    data={
      Array [
        Array [
          <styled.span>
            Patch owner
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-patchOwner"
      >
            Fred
      </span>,
        ],
        Array [
          <styled.span>
            Arrears status
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-status"
      >
            Active
      </span>,
        ],
        Array [
          <styled.span>
            Workwise reference
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-workwiseReference"
      >
            12345
      </span>,
        ],
        Array [
          <styled.span>
            Length of arrears
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-lengthOfArrears"
      >
            5 days (case started on 15/07/2018)
      </span>,
        ],
        Array [
          <styled.span>
            Payment reference number
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentReferenceNumber"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Collection rate (5W)
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentCollectionRate5w"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Collection rate (4M)
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentCollectionRate4m"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Gross rent
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-grossPayment"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Debit frequency
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentReferenceNumber"
      />,
        ],
        Array [
          <styled.span />,
          <span
            data-bdd="ArrearsDetails-arrearsCurrentBalance"
      >
            None
      </span>,
        ],
        Array [
          <Styled(WithTheme(Button))
            isText={true}
            to="/arrears-details/undefined/payment-plan/create"
      >
            Payment plan
      </Styled(WithTheme(Button))>,
          <span
            data-bdd="ArrearsDetails-paymentPlan"
      >
            <Unknown>
                  <styled.span>
                        Start date:
                  </styled.span>
                   
                  15/07/2018
                  <br />
                  <styled.span>
                        End date:
                  </styled.span>
                   
                  23/04/2017
                  <br />
                  <styled.span />
                   
                  £50.00 Every 2 Week(s)
                  <br />
                  <styled.span>
                        Type:
                  </styled.span>
                   
                  Foo
            </Unknown>
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/pause-arrears-case/create"
      >
            Pauses
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-pauses"
      >
            None
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/pause-arrears-case/create"
      >
            Draft pauses
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-draftPauses"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Referrals
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-referralStatus"
      >
            None
      </span>,
        ],
      ]
    }
    data-bdd="ArrearsDetailsTable"
    headerData={null}
    striped={true}
    tdWrap={true}
    topAlign={true}
  />
</styled.div>
`;

exports[`<ArrearsDetails /> renders correctly with referralSummaries 1`] = `
<styled.div>
  <Styled(styled.h3)>
    Arrears details
  </Styled(styled.h3)>
  <Table
    addExtraTd={false}
    addExtraTh={false}
    data={
      Array [
        Array [
          <styled.span>
            Patch owner
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-patchOwner"
      >
            Fred
      </span>,
        ],
        Array [
          <styled.span>
            Arrears status
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-status"
      >
            Active
      </span>,
        ],
        Array [
          <styled.span>
            Workwise reference
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-workwiseReference"
      >
            12345
      </span>,
        ],
        Array [
          <styled.span>
            Length of arrears
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-lengthOfArrears"
      >
            5 days (case started on 15/07/2018)
      </span>,
        ],
        Array [
          <styled.span>
            Payment reference number
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentReferenceNumber"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Collection rate (5W)
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentCollectionRate5w"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Collection rate (4M)
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentCollectionRate4m"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Gross rent
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-grossPayment"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Debit frequency
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentReferenceNumber"
      />,
        ],
        Array [
          <styled.span />,
          <span
            data-bdd="ArrearsDetails-arrearsCurrentBalance"
      >
            None
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/payment-plan/create"
      >
            Payment plan
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentPlan"
      >
            None
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/pause-arrears-case/create"
      >
            Pauses
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-pauses"
      >
            None
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/pause-arrears-case/create"
      >
            Draft pauses
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-draftPauses"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Referrals
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-referralStatus"
      >
            <Unknown>
                  Bob, 23/04/2017
                  <br />
            </Unknown>
      </span>,
        ],
      ]
    }
    data-bdd="ArrearsDetailsTable"
    headerData={null}
    striped={true}
    tdWrap={true}
    topAlign={true}
  />
</styled.div>
`;

exports[`<ArrearsDetails /> rent details renders correctly with rent details 1`] = `
<styled.div>
  <Styled(styled.h3)>
    Arrears details
  </Styled(styled.h3)>
  <Table
    addExtraTd={false}
    addExtraTh={false}
    data={
      Array [
        Array [
          <styled.span>
            Patch owner
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-patchOwner"
      >
            Fred
      </span>,
        ],
        Array [
          <styled.span>
            Arrears status
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-status"
      >
            Active
      </span>,
        ],
        Array [
          <styled.span>
            Workwise reference
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-workwiseReference"
      >
            12345
      </span>,
        ],
        Array [
          <styled.span>
            Length of arrears
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-lengthOfArrears"
      >
            5 days (case started on 15/07/2018)
      </span>,
        ],
        Array [
          <styled.span>
            Payment reference number
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentReferenceNumber"
      >
            12039
      </span>,
        ],
        Array [
          <styled.span>
            Collection rate (5W)
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentCollectionRate5w"
      >
            11%
      </span>,
        ],
        Array [
          <styled.span>
            Collection rate (4M)
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentCollectionRate4m"
      >
            10%
      </span>,
        ],
        Array [
          <styled.span>
            Gross rent
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-grossPayment"
      >
            £123.00
      </span>,
        ],
        Array [
          <styled.span />,
          <span
            data-bdd="ArrearsDetails-charge-undefined"
      >
            £5.00
      </span>,
        ],
        Array [
          <styled.span>
            Debit frequency
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentReferenceNumber"
      >
            Weekly
      </span>,
        ],
        Array [
          <styled.span />,
          <span
            data-bdd="ArrearsDetails-arrearsCurrentBalance"
      >
            -£1,230.00
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/payment-plan/create"
      >
            Payment plan
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-paymentPlan"
      >
            None
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/pause-arrears-case/create"
      >
            Pauses
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-pauses"
      >
            None
      </span>,
        ],
        Array [
          <styled.span
            isText={true}
            to="/arrears-details/undefined/pause-arrears-case/create"
      >
            Draft pauses
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-draftPauses"
      >
            None
      </span>,
        ],
        Array [
          <styled.span>
            Referrals
      </styled.span>,
          <span
            data-bdd="ArrearsDetails-referralStatus"
      >
            None
      </span>,
        ],
      ]
    }
    data-bdd="ArrearsDetailsTable"
    headerData={null}
    striped={true}
    tdWrap={true}
    topAlign={true}
  />
</styled.div>
`;
